### PR TITLE
modified OCF scripts for Neutron agents

### PR DESCRIPTION
**Pull request by xenolog**
port from 3.2.1

### CODE REVIEW

**Comment by aglarendil** [comment]
+1

File: deployment/puppet/neutron/files/ocf/neutron-agent-l3 [hunk 1]
```diff
@@ -96,17 +97,17 @@ creates a distributed neutron-l3-agent cluster.
...
 <content type="string" default="${OCF_RESKEY_binary_default}" />
 </parameter>
 
 <parameter name="config" unique="0" required="0">
 <longdesc lang="en">
-Location of the OpenStack Quantum Service (neutron-server) configuration file
+Location of the OpenStack OpenStack Router (neutron-server) configuration file
 </longdesc>
-<shortdesc lang="en">OpenStack L3 Server (neutron-server) config file</shortdesc>
+<shortdesc lang="en">OpenStack Router (neutron-server) config file</shortdesc>
```
Comment (xdel) [inline deployment/puppet/neutron/files/ocf/neutron-agent-l3:110]: server? router?

File: deployment/puppet/neutron/files/ocf/neutron-agent-l3 [hunk 2]
```diff
@@ -253,42 +267,51 @@ neutron_l3_agent_validate() {
...
+    if [ "xxx$pid" == "xxx" ] ; then
+        ocf_log warn "OpenStack Neutron agent '$OCF_RESKEY_binary' not running."
+        return $OCF_NOT_RUNNING
+    fi
+    #ocf_log debug "PID='$pid'"
 
-        pid=`pgrep -f ${OCF_RESKEY_binary}`
+    # Check PID file and create if need
+    if [ ! -f $OCF_RESKEY_pid ] ; then
+        ocf_log warn "OpenStack Neutron agent (${OCF_RESKEY_binary}) was run, but no PID file found."
```
Comment (xdel) [inline deployment/puppet/neutron/files/ocf/neutron-agent-l3:285]: check grammar

File: deployment/puppet/neutron/files/ocf/neutron-agent-l3 [hunk 3]
```diff
@@ -253,42 +267,51 @@ neutron_l3_agent_validate() {
...
+    f_pid=`cat $OCF_RESKEY_pid | tr '\n' ' ' | awk '{print $1}'`
+    if [ "xxx$pid" == "xxx$f_pid" ]; then
+        return $OCF_SUCCESS
     fi
 
-    ocf_run -warn kill -s 0 $pid
-    rc=$?
-    if [ $rc -eq 0 ]; then
+    # at this point we have PID file and PID from it
+    # defferents with PID from `pgrep...`
```
Comment (xdel) [inline deployment/puppet/neutron/files/ocf/neutron-agent-l3]: differents

File: deployment/puppet/neutron/files/ocf/neutron-agent-l3 [hunk 4]
```diff
@@ -350,16 +348,17 @@ neutron_l3_agent_start() {
...
+        if [ $rc -ne $OCF_NOT_RUNNING ] ; then
             ocf_log err "OpenStack neutron-l3-agent start failed"
             exit $OCF_ERR_GENERIC
         fi
-        sleep 1
+        sleep 3
     done
 
-    sleep 13 ; q-agent-cleanup.py --agent=l3 --reschedule --remove-dead 2>&1 >> /var/log/neutron/rescheduling.log
+    # detach deffered rescheduling procedure
```
Comment (xdel) [inline deployment/puppet/neutron/files/ocf/neutron-agent-l3]: deferred

File: deployment/puppet/neutron/files/ocf/neutron-agent-l3 (hunk 1 above)
Comment (xenolog) [inline deployment/puppet/neutron/files/ocf/neutron-agent-l3:110]: agent is a Router

File: deployment/puppet/neutron/files/ocf/neutron-agent-l3 (hunk 2 above)
Comment (xenolog) [inline deployment/puppet/neutron/files/ocf/neutron-agent-l3:285]: what do you mean?

**Pull request merged**